service/debugger/debugger.go

```diff
@@ -434,6 +434,15 @@ func (d *Debugger) Restart(rerecord bool, pos string, resetArgs bool, newArgs []
 	}
 	var p *proc.Target
 	var err error
+
+	// TODO executeKind from command.go
+	if rebuild && d.canRestart() {
```

Comment:
The code above has a `if !d.canRestart() { return ... }` why do you need to check this here too?

Also the loop that sets breakpoints below should discard the ones that can't be set by file/line.

Reply:
You are totally right. I removed the d.canRestart.

Can you please clarify a little more the loop part? I don't really understand what you mean.

Reply:
In this function there's a loop, `for _, oldBp := ...` that restores breakpoints. It either does it by file:line (`if len(oldBp).File > 0`) or by address (`else`). We shouldn't try to set them by address because addresses have changed after rebuilding.

Reply:
Thanks for the explanation! I updated the loop to avoid the case